modules/catalog-compaction/src/integrationTest/java/org/apache/ignite/internal/catalog/compaction/ItCatalogCompactionTest.java

```diff
@@ -51,14 +51,13 @@
 import org.apache.ignite.network.ClusterNode;
 import org.apache.ignite.tx.Transaction;
 import org.apache.ignite.tx.TransactionOptions;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 /**
  * Integration tests to verify catalog compaction.
  */
-class ItCatalogCompactionTest extends ClusterPerClassIntegrationTest {
+class ItCatalogCompactionTest extends ClusterPerTestIntegrationTest {
```

Comment:
Due to personal discussion there was found that the drastic reason relates to [IGNITE-23753](https://issues.apache.org/jira/browse/IGNITE-23753).

If we'll add replica state on assertion at the revision `1fe7f5ff` we could see then `ASSIGNED` state for already removed replicas. The problem is similar to [IGNITE-23753](https://issues.apache.org/jira/browse/IGNITE-23753), but reason differ: we have 2 places in `TableManager` where `stopReplica` raw calls are done:
1. [IGNITE-23753](https://issues.apache.org/jira/browse/IGNITE-23753)'s issue on Ignite node stopping process;
2. on drop table process: `onLwmChanged`  -> `destroyTableLocally` -> `stopAndDestroyPartition` -> `stopPartition` -> `stopReplica`.

In both cases we call `stopReplica` instead `weakStopReplica` with a particular reason and replica's states handling. It leads to wrong states and possible memory leak because `ASSIGNED` states shouldn't be deleted. 

The issue should be investigated in [IGNITE-23753](https://issues.apache.org/jira/browse/IGNITE-23753), for now test class after 20 repeated runs is green locally due to no-op handling on null-value replica future. But bug with states is confirmed.